src/core/patch.rkt

```diff
@@ -71,16 +71,25 @@
                               #;(exp ,exp-x ,log-x)
                               #;(log ,log-x ,exp-x))))
 
-(define (taylor-alt altn)
-  (define expr (prog->spec (alt-expr altn)))
+(define (taylor-alts altns)
+  (define exprs
+    (for/list ([altn (in-list altns)])
+      (prog->spec (alt-expr altn))))
+  (define free-vars (map free-variables exprs))
+  (define vars (list->set (append* free-vars)))
+
   (reap [sow]
-        (for* ([var (free-variables expr)]
+        (for* ([var (in-set vars)]
                [transform-type transforms-to-try])
           (match-define (list name f finv) transform-type)
-          (define timeline-stop! (timeline-start! 'series (~a expr) (~a var) (~a name)))
-          (define genexpr (approximate expr var #:transform (cons f finv)))
-          (for ([_ (in-range (*taylor-order-limit*))])
-            (sow (alt (genexpr) `(taylor ,name ,var) (list altn) '())))
+          (define timeline-stop! (timeline-start! 'series (~a exprs) (~a var) (~a name)))
+          (define genexprs (approximate exprs var #:transform (cons f finv)))
+          (for ([genexpr (in-list genexprs)]
+                [altn (in-list altns)]
+                [fv (in-list free-vars)]
+                #:when (member var fv)) ; check whether var exists in expr at all
+            (for ([_ (in-range (*taylor-order-limit*))])
+              (sow (alt (genexpr) `(taylor ,name ,var) (list altn) '()))))
```

Comment:
Here we already store a `prev altn` for an approximation, which makes `approx->prev` hash useless, because we can access an original alternative through the output-alternative